go/vt/vtgate/planbuilder/abstract/operator.go

```diff
@@ -132,6 +132,93 @@ func crossJoin(exprs sqlparser.TableExprs, semTable *semantics.SemTable) (Operat
 	return output, nil
 }
 
+func getSelect(s sqlparser.SelectStatement) *sqlparser.Select {
+	switch s := s.(type) {
+	case *sqlparser.Select:
+		return s
+	case *sqlparser.ParenSelect:
+		return getSelect(s.Select)
+	default:
+		return nil
+	}
+}
+
+// CreateOperatorFromSelectStmt creates an operator tree that represents the input SELECT or UNION query
+func CreateOperatorFromSelectStmt(selStmt sqlparser.SelectStatement, semTable *semantics.SemTable) (Operator, error) {
+	switch node := selStmt.(type) {
+	case *sqlparser.Select:
+		return CreateOperatorFromSelect(node, semTable)
+	case *sqlparser.Union:
+		return createOperatorFromUnion(node, semTable)
+	case *sqlparser.ParenSelect:
+		return CreateOperatorFromSelectStmt(node.Select, semTable)
+	}
```

Comment:
`CreateOperatorFromSelect` and `CreateOperatorFromSelectStmt` can be non-exported, the same way we did for `createOperatorFromUnion`.